src/Sylius/Bundle/ChannelBundle/Collector/ChannelCollector.php

```diff
@@ -35,12 +35,12 @@ public function __construct(
 
         $this->data = [
             'channel' => null,
-            'channels' => $channelRepository->findAll(),
+            'channels' => array_map([$this, 'pluckChannel'], $channelRepository->findAll()),
             'channel_change_support' => $channelChangeSupport,
         ];
     }
 
-    public function getChannel(): ?ChannelInterface
+    public function getChannel(): array
```

Comment:
It should be `?array`, in case the current channel could not be resolved.

Reply:
good point, fixed